asreview/webapp/src/Components/ElasGame.js

```diff
@@ -50,77 +86,301 @@ const GameStyle = styled(Box)(({ theme }) => ({
   },
 }));
 
+// Function to shuffle images for the game
+function shuffleImages(images) {
+  let shuffledImages = [...images];
+  for (let i = shuffledImages.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [shuffledImages[i], shuffledImages[j]] = [
+      shuffledImages[j],
+      shuffledImages[i],
+    ];
+  }
+  return shuffledImages;
+}
+
+// Function to prepare the initial shuffle of images based on game mode
+const initialShuffle = (mode) => {
+  const imagesToUse = mode === "simple" ? images.slice(0, 8) : images;
+  const duplicatedImages = imagesToUse.concat(imagesToUse);
+  return shuffleImages(duplicatedImages);
+};
+
+// Main component
 const ElasGame = (props) => {
-  const [imagesArray, setImagesArray] = useState([]);
-  const [paperSelected, setPaperSelected] = useState([]);
-  const [paperSelectedIds, setPaperSelectedIds] = useState([]);
+  // Define initial state
+  const initialState = {
+    mode: "simple",
+    cheatMode: 'inactive',
+    gameStarted: false,
+    imagesArraySimple: initialShuffle("simple"),
+    imagesArrayExpert: initialShuffle("expert"),
+    imagesArray: [],
+    isExpertCompleted: false,
```

Comment:
I prefer not to have different behavior for different games. `isComplete: false`